### PR TITLE
June destination

### PR DESCRIPTION
**Pull request by ferrucc-io**
_A summary of your pull request, including the what change you're making and why._

## Testing

Tested all endpoints and it seems like all data gets through correctly:

Screenshot 2023-03-14 at 19 00 56

_Include any additional information about the testing you have completed to
ensure your changes behave as expected. For a speedy review, please check
any of the tasks you completed below during your testing._

- [x] Added [unit tests](https://github.com/segmentio/action-destinations/blob/main/docs/testing.md#local-end-to-end-testing) for new functionality
- [x] Tested end-to-end using the [local server](https://github.com/segmentio/action-destinations/blob/main/docs/testing.md#local-end-to-end-testing)
- [ ] [Segmenters] Tested in the staging environment

### CODE REVIEW

**Comment by joe-ayoub-segment** [comment]
hi @ferrucc-io , acknowledging this PR. 
As with any new Integration (or rewrite of an existing Integration) I'd like to ask you to schedule a call with me so we can run through the use-cases for this Integration. I'd also like to understand the types of events which will be sent from Segment to June, as well as the identifiers to be included in the events. 

Would you be OK scheduling a call with me via [this Calendly link](http://calendly.com/joe_ayoub/integration-workshop) please?

**Comment by ferrucc-io** [comment]
Just scheduled a call for Thursday

File: packages/destination-actions/src/destinations/june/track/index.ts [hunk 1]
```diff
@@ -0,0 +1,77 @@
...
+  perform: (request, { payload }) => {
+    return request('https://api.june.so/api/track', {
+      method: 'POST',
+      json: {
+        event: payload.event,
+        properties: payload.properties,
+        timestamp: payload.timestamp,
+        userId: payload.userId,
+        anonymousId: payload.anonymousId,
+        context: payload.context,
```
Comment (joe-ayoub-segment) [inline packages/destination-actions/src/destinations/june/track/index.ts:70]: recommend a new field specifically for groupId. You could then merge in the value for this field (in the perform function) into the context object.

**Review by joe-ayoub-segment — CHANGES_REQUESTED** [review]

**Review by joe-ayoub-segment — APPROVED** [review]

**Comment by joe-ayoub-segment** [comment]
June API is practically identical to the Segment API. 
Each Action is very simple and has presets available. 
Only recommendation (which will be done in a further PR) would be to add a specific field for groupId for both track() and page() calls.

**Comment by ferrucc-io** [comment]
Interestingly I get an empty diff
Screenshot 2023-03-16 at 17 57 47

**Comment by joe-ayoub-segment** [comment]
I think the issue relates to timestamps being executed in different timezones. When you run the tests they pass, but our CI is running in a different timezone....

**Comment by ferrucc-io** [comment]
@joe-ayoub-segment just fixed it by running

```
TZ=UTC yarn jest --testPathPattern='./packages/destination-actions/src/destinations/june' --updateSnapshot
```

Basically snapshots need to have the correct timezone and it can be configured as an env variable

**Comment by joe-ayoub-segment** [comment]
hi @ferrucc-io , the tests still fail for me unfortunately. 
These are the same tests which are run by CI, so it won't be possible to specify a variable when running. 

Is there another way you can fix this test?

**Comment by joe-ayoub-segment** [comment]
Hi @ferrucc-io , 
I pushed a change to your fork which updated this PR. The change forces the tests to ignore the timestamp. 
PR is now merged and should be deployed this week.

**Comment by ferrucc-io** [comment]
Awesome thanks for your help!

**Comment by joe-ayoub-segment** [comment]
FYI @ferrucc-io the deploy didn't complete yesterday. Expecting to deploy this today.

**Pull request merged**